metals/src/main/scala/scala/meta/internal/pantsbuild/BloopPants.scala

```diff
@@ -262,9 +262,38 @@ private class BloopPants(
     )
     val isBaseDirectory =
       projects.iterator.filter(_.sources.nonEmpty).map(_.directory).toSet
+
+    def existsProjectDefinedAtRoot(
+        projects: Seq[C.Project],
+        root: AbsolutePath
+    ): Boolean =
+      projects.exists(p =>
+        p.sources.exists(_.getParent == root.toNIO)
+          || p.sourcesGlobs.exists(_.exists(_.directory == root.toNIO))
```

Comment:
Is this condition necessary? The `.directory` field is always `baseDirectory` https://github.com/scalameta/metals/blob/775d2bff9c37acab065a9a465e6984db7ad1c03e/metals/src/main/scala/scala/meta/internal/pantsbuild/PantsGlobs.scala#L34

Reply:
right

Reply:
changed the whole condition `projects.exists(_.directory == root.toNIO)`

Reply:
This case is already covered by `isBaseDirectory`, which contains `Project.directory` entries. I'm not sure this condition needs to exist.

I suspect the problem with the synthetic modules approach is that we're generating modules with a resource directory. I feel like we need a different approach. I think it's best to close this for now and I will try to think of something else.